src/components/Consent.js

```diff
@@ -1,44 +1,46 @@
 import { h } from 'preact'
-import { useI18n, useSoftkey, useOnlineStatus } from 'hooks'
+import { useRef, useEffect } from 'preact/hooks'
+import { useI18n, useSoftkey, useOnlineStatus, useScroll } from 'hooks'
 import { grantConsent, goto } from 'utils'
-import { OfflinePanel } from 'components'
 
-export const Consent = () => {
+export const Consent = ({ close }) => {
   const i18n = useI18n()
   const isOnline = useOnlineStatus()
+  const containerRef = useRef()
+  const [scrollDown, scrollUp] = useScroll(containerRef, 10, 'y')
 
   const onAgree = () => {
     grantConsent()
-    goto.search()
+    close()
   }
 
-  const softkeyConfig = {
-    offline: {
-      backspace: () => { window.exit() }
-    },
-    online: {
-      center: i18n('consent-softkeys-agree'),
-      onKeyCenter: onAgree,
-      left: i18n('consent-softkeys-terms'),
-      onKeyLeft: goto.termsOfUse,
-      right: i18n('consent-softkeys-policy'),
-      onKeyRight: goto.privacyPolicy,
-      backspace: () => { window.exit() }
+  useSoftkey('ConsentMessage', {
+    center: i18n('consent-softkeys-agree'),
+    onKeyCenter: onAgree,
+    left: i18n('consent-softkeys-terms'),
+    onKeyLeft: goto.termsOfUse,
+    right: i18n('consent-softkeys-policy'),
+    onKeyRight: goto.privacyPolicy,
+    onKeyArrowDown: scrollDown,
+    onKeyArrowUp: scrollUp,
+    onKeyBackspace: () => { window.close() }
+  })
+
+  useEffect(() => {
+    if (!isOnline) {
+      close()
     }
-  }
-  useSoftkey('ConsentMessage', softkeyConfig[isOnline ? 'online' : 'offline'], [isOnline], true)
+  }, [isOnline])
```

Comment:
Reasoning and test I did for this here:

* installed app on device
* wifi is on
* open app and get to consent screen as expected
* leave wifi on and app open and walked two floors down the stairway until connection was lost (rather than closing the app and turning wifi off manually)
* as connection was lost, the app closed consent screen and displayed offline panel

Reply:
the above step is valid and make sense, and user can continue with

1. on the offline screen, type query (the app doesn't send the api request)
2. once the app goes online, the app shows consent screen correctly, but in the background it loads search data 

😅

Reply:
Good catch! What about https://github.com/wikimedia/wikipedia-kaios/pull/309/commits/cc18e0ed7bce2a68aa2d53ac2ed4c2844e0bdf77 to disable input in that edge case?